server/implementation/src/main/java/io/smallrye/graphql/scalar/custom/CustomStringScalar.java

```diff
@@ -0,0 +1,58 @@
+package io.smallrye.graphql.scalar.custom;
+
+import java.lang.reflect.Type;
+
+import jakarta.json.JsonValue.ValueType;
+import jakarta.json.bind.serializer.DeserializationContext;
+import jakarta.json.bind.serializer.JsonbDeserializer;
+import jakarta.json.bind.serializer.JsonbSerializer;
+import jakarta.json.bind.serializer.SerializationContext;
+import jakarta.json.stream.JsonGenerator;
+import jakarta.json.stream.JsonParser;
+
+import io.smallrye.graphql.spi.ClassloadingService;
+
+/**
+ * A base class for all CustomScalars that are based on GraphQL's String.
+ */
+public interface CustomStringScalar {
```

Comment:
I've been thinking that this is actually public API, so maybe this (and the other two) interfaces should be moved to the  `io.smallrye.graphql.api` package. The API module doesn't depend on JSON-B, so you'll also have to move out the `SERIALIZER` and `DESERIALIZER` fields somewhere else, I guess they could be directly in `io.smallrye.graphql.json.JsonBCreator` for instance?